numpy/lib/npyio.py

```diff
@@ -1006,7 +1020,18 @@ def split_line(line):
                 except ValueError:
                     # Unused converter specified
                     continue
-            converters[i] = conv
+            if byte_converters:
+                # converters may use decode to workaround numpy's oldd behaviour,
+                # so encode the string again before passing to the user converter
+                def tobytes_first(x, conv):
```

Comment:
This seems to redefine `asbytes` from `py3k.py`; can one just use that? I.e., something like
```
converters[i] = lambda x: conv(asbytes(x))
```